document/src/main/java/com/yahoo/document/update/TensorModifyUpdate.java

```diff
@@ -29,8 +30,8 @@ public TensorModifyUpdate(Operation operation, TensorFieldValue tensor) {
     }
 
     private void verifyCompatibleType(TensorType type) {
-        if (type.dimensions().stream().anyMatch(dim -> dim.isIndexed()) ) {
-            throw new IllegalArgumentException("Tensor type '" + type + "' is not compatible as it contains some indexed dimensions");
+        if (type.rank() > 0 && type.dimensions().stream().noneMatch(dim -> dim.isMapped()) ) {
```

Comment:
Class comment should be updated now that operation is supported for mixed tensors.